source/pymongo-to-async-guide.txt

```diff
@@ -0,0 +1,328 @@
+.. _pymongo-to-async-guide:
+
+====================================
+Switch from {+driver-short+} to {+driver-async+}
+====================================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: asyncronous, refactor, migration
+
+Overview
+--------
+
+The {+driver-async+} driver is a unification of {+driver-short+} and the `Motor
+library <https://www.mongodb.com/docs/drivers/motor/>`__. In this guide, you can
+identify the changes you must make to switch from {+driver-short+} to
+{+driver-async+}.
+
+Switch From {+driver-short+}
+-------------------
+
+The {+driver-async+} driver behaves similarly to {+driver-short+}, but
+all methods that perform network operations are coroutines and must be awaited.
+To switch from {+driver-short+} to {+driver-async+}, you must update your code
+in the following ways:
+
+- Replace all uses of ``MongoClient`` with ``AsyncMongoClient``.
+- Add the ``await`` keyword to all asynchronous method calls.
+- If an asynchronous method is called within a function, mark the function as ``async``.
```

Comment:
It's unclear that this last point is for functions that the user defines.

Reply:
Good call. Updated